components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -636,16 +731,92 @@ public void updateConsumerApplication(OAuthConsumerAppDTO consumerAppDTO) throws
             oauthappdo.setTokenRevocationWithIDPSessionTerminationEnabled(consumerAppDTO
                     .isTokenRevocationWithIDPSessionTerminationEnabled());
             oauthappdo.setTokenBindingValidationEnabled(consumerAppDTO.isTokenBindingValidationEnabled());
-            oauthappdo.setTokenEndpointAuthMethod(consumerAppDTO.getTokenEndpointAuthMethod());
-            oauthappdo.setTokenEndpointAuthSignatureAlgorithm(
-                    consumerAppDTO.getTokenEndpointAuthSignatureAlgorithm());
-            oauthappdo.setSectorIdentifierURI(consumerAppDTO.getSectorIdentifierURI());
-            oauthappdo.setIdTokenSignatureAlgorithm(consumerAppDTO.getIdTokenSignatureAlgorithm());
-            oauthappdo.setRequestObjectSignatureAlgorithm(consumerAppDTO.getRequestObjectSignatureAlgorithm());
+
+            String tokenEndpointAuthMethod = consumerAppDTO.getTokenEndpointAuthMethod();
+            if (StringUtils.isNotEmpty(tokenEndpointAuthMethod)) {
+                if (enforceFAPIDCR) {
+                    validateFAPITokenAuthMethods(tokenEndpointAuthMethod);
+                } else {
+                    filterTokenEndpointAuthMethods(tokenEndpointAuthMethod);
+                }
+                oauthappdo.setTokenEndpointAuthMethod(tokenEndpointAuthMethod);
+            }
+
+            String tokenEndpointAuthSignatureAlgorithm = consumerAppDTO.getTokenEndpointAuthSignatureAlgorithm();
+            if (StringUtils.isNotEmpty(tokenEndpointAuthSignatureAlgorithm)) {
+                if (enforceFAPIDCR) {
+                    validateFAPISignatureAlgorithms(tokenEndpointAuthSignatureAlgorithm);
+                } else {
+                    filterSignatureAlgorithms(tokenEndpointAuthSignatureAlgorithm,
+                            OAuthConstants.TOKEN_EP_SIGNATURE_ALG_CONFIGURATION);
+                }
+                oauthappdo.setTokenEndpointAuthSignatureAlgorithm(tokenEndpointAuthSignatureAlgorithm);
+            }
+
+            if (StringUtils.isNotEmpty(consumerAppDTO.getSubjectType())) {
+                OAuthConstants.SubjectType subjectType = OAuthConstants.SubjectType.fromValue(
+                        consumerAppDTO.getSubjectType());
+                if (subjectType == null) {
+                    consumerAppDTO.setSubjectType(OIDCClaimUtil.getDefaultSubjectType().toString());
```

Comment:
with StringUtils.isNotEmpty(consumerAppDTO.getSubjectType()) if the subject type is not set then this code block will also not be reached right? so we should check and set a default if consumerAppDTO.getSubjectType() is empty

Reply:
fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/e4cfcc774433861b5de411a40c83fea6f9ca5d38